### ISSUE TITLE
After migration from very old spreed installation unrelated messages are listed in newly created rooms

### ISSUE THREAD

**Issue by robfuscator** [nextcloud/spreed#11299]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

I recently migrated to Nextcloud AIO. Since it comes with a TURN server & the like, I thought I'd give Nextcloud Talk another try (I used it on version 11 for the last time, IIRC, but disabled it again since setting up an own TURN server seemed like an overkill for me at that time).

The migration first failed, as described here: https://github.com/nextcloud/spreed/issues/7140#issuecomment-1097729834
But after running the SQL queries mentioned in the issue, I could install the app.

Now my problem is, that there seem to be some "zombie" chatrooms and messages in the database which interfere with the newly installed app. When I open a new room, I sometimes see messages I or other people exchanged years ago or get old messages from the "Talk Updates" room displayed in other rooms.

Seems pretty messed up to me ^^. Is this a known problem, and if not, is there any way to just purge everything related to spreed and start over again? I luckily don't have any production data in spreed. Any help is highly appreciated!

And big shout outs to the whole Nextcloud team, you guys are doing an incredibly nice job :smile:.

---

## Steps to reproduce
1. Migrate from a very old spreed installation
2. Create new room

### Expected behaviour
Have a clean room

### Actual behaviour
Old messages from unrelated rooms are shown

## Talk app

**Talk app version:** Up-to-date AIO

**Comment by Antreesy** [comment]
Hello there!

Sounds to me like an overlap of tokens with old rooms. There is no occ command to list all existent rooms, but I suppose, if you could check them in the database, the command `occ talk:room:delete <token>` should do the trick. Please let us know later about the results

**Comment by robfuscator** [comment]
Thank you for your reply! Can you give me a hint about where I could find the rooms in the database? I am familiar with connecting to it and running SQL queries but don't know about the table structure. That would be highly appreciated :)

**Comment by robfuscator** [comment]
Hmm, that's confusing. I managed to find the table `oc_talk_rooms` by scrolling through the output of the psql `\dt` command. There were indeed 3 tables listed. I tried to delete them using the `occ` command as you suggested, which failed with `Room is no group call.`. I then decided to remove the Talk app again and drop all related tables (again, as suggested here https://github.com/nextcloud/spreed/issues/7140#issuecomment-1097729834).

I tried creating some more rooms, at the behavior really seems random to me. Sometimes I just get a "clean", empty room. But sometimes there are again messages which I would expect in the Talk Updates room. I'll attach a picture. What I also find very confusing is that it says "XY created the conversation", but XY being not my name but that of another user on my instance. Such messages also seemingly sporadically appear throughout the Talk Updates in the room, sometimes with my username, sometimes with another.

![image](https://github.com/nextcloud/spreed/assets/38418563/ab23e7d4-e2aa-43fc-a770-17575cc2c2b6)

Implies to me that there must be more data stored somewhere related to Talk. Do you have any idea how to investigate this?

**Comment by Antreesy** [comment]
I'd wait for @nickvergessen if he has any thoughts on it. Couldn't say from the frontend side, what is wrong here. 

At least the `Talk updates` room, after the complete cleanup should be recreated and started from a time you logged in:

![image](https://github.com/nextcloud/spreed/assets/93392545/16557364-396c-4954-ac98-df6a495547f6)

**Comment by nickvergessen** [comment]
`oc_comments` table holds the talk chat messages and comments of other apps.
You should delete all the rows with `object_type ='chat'` from it and all content from all `oc_talk_*` tables to reset it correctly. Otherwise room IDs will be reused leaking messages into rooms with the same ID now.

**Comment by robfuscator** [comment]
Thank you both. There were 3 rows left in the oc_comments table, related to Talk. Is it possible that there were more but by creating and deleting rooms during my "investigation" they got removed already? Because I'm sure I've seen more than 3 such random old messages.

After deleting, I've created around 20 rooms and the rooms were always clean. Only thing I've noticed was this https://github.com/nextcloud/spreed/issues/5763 which seems to be known.

I can't see the Talk Updates room, but I think that's because I've deleted it for my user. I created a test user and that one can see the room.

Would it be helpful to create some kind of "last resort" FAQ post somewhere explaining how to do a complete reset of the app? I've seen quite some issues and and discussions about migrations from older installations not working.
